File: src/console/command/impl/ClearCommand.php
<?php

namespace pocketcloud\cloud\console\command\impl;

use pocketcloud\cloud\console\command\Command;
use pocketcloud\cloud\console\command\sender\ICommandSender;
use pocketcloud\cloud\console\command\SubCommand;
use pocketcloud\cloud\console\log\logger\cache\LogMessagesCache;
use pocketcloud\cloud\util\TerminalUtils;

final class ClearCommand extends Command {

    public function __construct() {
        parent::__construct("clear", "Clears the console", ["cls", "purge"]);
    }

    public function run(ICommandSender $sender, string $label, array $args, ?SubCommand $subCommand, array $flags): bool {
        LogMessagesCache::clear();
        TerminalUtils::clearConsole();
        return true;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/impl/terminal/ClearCommand.java
package de.pocketcloud.cloud.console.command.impl.terminal;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.command.Command;
import de.pocketcloud.cloud.console.command.ctx.CommandContext;
import de.pocketcloud.cloud.console.command.desc.CommandDescription;
import de.pocketcloud.cloud.console.command.sender.CommandSender;

@CommandDescription(name = "clear", description = "Clears the console", aliases = {"cls"})
public final class ClearCommand extends Command {

    @Override
    public void prepare() {}

    @Override
    public boolean run(CommandSender sender, CommandContext ctx) {
        PocketCloud.instance().console().clear();
        return true;
    }
}